### PR TITLE
add support for missing spring list properties

### PR DESCRIPTION
**Pull request by zeitlinger**

### CODE REVIEW

File: instrumentation/spring/spring-boot-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 1]
```diff
@@ -419,8 +419,8 @@
     },
     {
       "name": "otel.instrumentation.logback-appender.experimental.capture-mdc-attributes",
-      "type": "java.util.List<java.lang.String>",
-      "description": "MDC attributes to capture. Use the wildcard character <code>*</code> to capture all attributes."
+      "type": "java.lang.String",
```
Comment (zeitlinger) [inline instrumentation/spring/spring-boot-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:422]: @jeanbisutti this property is used by https://github.com/open-telemetry/opentelemetry-java-instrumentation/blob/5b21872533968c9fdb0347b3a8526f84aed19174/instrumentation/spring/spring-boot-autoconfigure/src/main/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/instrumentation/logging/LogbackAppenderInstaller.java#L19 - and I don't know if we can access `@CofigurationProperties` at that stage yet

**Comment by zeitlinger** [comment]
@jeanbisutti please review

File: instrumentation/spring/spring-boot-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json (hunk 1 above)
Comment (jeanbisutti) [reply instrumentation/spring/spring-boot-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:422]: Not sure what you mean.

From what I observe, `@Value` does not seem to work in `LogbackAppenderApplicationListener`. 

`org.springframework.boot.context.logging.LoggingApplicationListener` does not use `@Value` .

Comment (jeanbisutti) [inline instrumentation/spring/spring-boot-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:422]: Why it is no longer a list of String?

**Review by jeanbisutti — COMMENTED** [review]
To help make the release note, could you please list the added properties in the description?

Are we sure that all the added properties are working? Perhaps it may be worth adding automatic tests.

File: instrumentation/spring/spring-boot-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json (hunk 1 above)
Comment (zeitlinger) [reply instrumentation/spring/spring-boot-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:428]: It never technically was - it's not possible to use yaml lists

File: instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/properties/SpringConfigPropertiesTest.java [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+            })
+        .map(Arguments::of);
+  }
+
+  @ParameterizedTest
+  @MethodSource("listProperties")
+  @DisplayName("should map list from application.yaml list")
+  void listsShouldWorkWithYaml(String key) {
+    List<String> expected;
+    if (key.equals("otel.propagators")) {
```
Comment (jeanbisutti) [inline instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/properties/SpringConfigPropertiesTest.java:115]: The test could be easier to read by removing logic from the test.

File: instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/properties/SpringConfigPropertiesTest.java [hunk 3]
```diff
@@ -0,0 +1,173 @@
...
+              "otel.instrumentation.http.server.capture-response-headers",
+              "otel.instrumentation.http.known-methods",
+            })
+        .map(Arguments::of);
+  }
+
+  @ParameterizedTest
+  @MethodSource("listProperties")
+  @DisplayName("should map list from application.yaml list")
+  void listsShouldWorkWithYaml(String key) {
```
Comment (jeanbisutti) [inline instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/properties/SpringConfigPropertiesTest.java:113]:
```suggestion
  // See the application.yaml file
  void listsShouldWorkWithYaml(String key) {
```

File: instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/properties/SpringConfigPropertiesTest.java (hunk 2 above)
Comment (zeitlinger) [reply instrumentation/spring/spring-boot-autoconfigure/src/test/java/io/opentelemetry/instrumentation/spring/autoconfigure/internal/properties/SpringConfigPropertiesTest.java]: fixed

**Review by jeanbisutti — APPROVED** [review]

**Comment by zeitlinger** [comment]
> Did these list properties work at all previously or did the work in some different way? 

Old:

```yaml
prop: a,b
```

New:

```yaml
prop: a,b
```

OR

```yaml
prop: [a,b]
```

> If the worked in some different way is this change backwards compatible?

yes

**Review by laurit — APPROVED** [review]

**Pull request merged**